pkg/codegen/go/gen_program_ternaries.go

```diff
@@ -0,0 +1,64 @@
+package gen
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/hcl2/model"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/hcl2/syntax"
+)
+
+type ternaryTemp struct {
+	Name  string
+	Value *model.ConditionalExpression
+}
+
+func (tt *ternaryTemp) Type() model.Type {
+	return tt.Value.Type()
+}
+
+func (tt *ternaryTemp) Traverse(traverser hcl.Traverser) (model.Traversable, hcl.Diagnostics) {
+	return tt.Type().Traverse(traverser)
+}
+
+func (tt *ternaryTemp) SyntaxNode() hclsyntax.Node {
+	return syntax.None
+}
+
+type tempAllocator struct {
+	temps []*ternaryTemp
+}
+
+func (ta *tempAllocator) allocateExpression(x model.Expression) (model.Expression, hcl.Diagnostics) {
+	var temp *ternaryTemp
+	switch x := x.(type) {
+	case *model.ConditionalExpression:
+		cond, _ := ta.allocateExpression(x.Condition)
+		t, _ := ta.allocateExpression(x.TrueResult)
+		f, _ := ta.allocateExpression(x.FalseResult)
```

Comment:
These shouldn't be necessary--you should be able to do `x.Condition, _ = ta.allocateExpression(x.Condition)`. Is that not working?

Reply:
ah yes, forgot to drop the `:` when I tried to make the change the first time.